source/administration/plugins.rst

```diff
@@ -12,12 +12,68 @@ Consider using a plugin in the following scenarios:
+2. Fill in Plugin settings as required
+3. At the top of the page, set **Enable Plugin** setting to ``True``.  If this flag is not switched on, the plugin will never become active.
+4. Test out the plugin as needed
+
+
+Upgrading Plugins
+~~~~~~~~~~~~~~~~~
+
+In v5.16, the Marketplace only supports the installation of new plugins.  To upgrade a plugin, you will need to manually update it by downloading the binary file from the GitHub repository and uploading it in the **System Console > Plugin management** section.  In v5.18 (Dec 2019), the Marketplace will allow you to upgrade a plugin on demand when a new version becomes available.
+
+
+Marketplace Server
+-----
+
+There are two Marketplace settings in the **System Console > Plugin management** section:
+
+.. image:: https://user-images.githubusercontent.com/915956/66892854-94660d80-efa1-11e9-805c-a85223d43a07.png
+
+- **Enable Marketplace** Turns the Marketplace UI on or off for all users (including sysadmins)
+- **Marketplace URL** The location of the Marketplace Server you want to query for new plugins. Mattermost hosts a Plugin Marketplace for the community and is the default value for this field.  You can setup your own Marketplace server, see below.
+
+When you are first presented with the Marketplace, your Mattermost server will attempt to contect the Marketplace Server,run by Mattermost, and will be returned a list of available plugins that are appropriate based on the server version that is currently running.  Only your server version and search query is passed over to Mattermost Marketplace, we retain an anonymized record for product analytics whenever a new plugin is installed, unless you have opted out of `Telemetry <https://docs.mattermost.com/administration/telemetry.html>`__ previously. The `Plugin marketplace server code <https://github.com/mattermost/mattermost-marketplace>`__ is available as an open-source project and can be used to setup your own private Marketplace if desired.  
+
+
+Mattermost Marketplace
+~~~~~~~~~~~~~~~~~~~~~~~
+
+The `Mattermost Plugin Marketplace <https://github.com/mattermost/mattermost-marketplace>`__ is a service run by Mattermost that contains listings of plugins that we have reviewed and in many cases built. In the future, we plan to include community developed plugins that will be labeled differently than "mattermost developed" plugins.  We plan to include settings that would restrict which types of plugins you can install.  Comments in our forum are welcome as we develop this feature further. 
+
+Mattermost Integration Directory
+-----
+
+There are many ways to integrate Mattermost aside from plugins, and we have created a directory of integrations "recipes", some of which are scripts, plugins, or instructions on how to connect Mattermost with your Enterprise systems. Many are sourced from our community of customers.  You can browse the directory at `https://integrations.mattermost.com/ <https://integrations.mattermost.com/>`__.
```

Comment:
This paragraph seams misplaced here. The whole page is about plugin and this paragraph is talking about integrations in general. What do you think about moving it into https://docs.mattermost.com/developer/integration-faq.html ?

Reply:
This link was there before, and I think because we don't have ANY community plugins listed yet, if someone wants to find other integrations....that directory is the best spot for right now.  And many are plugins that cannot be found on the marketplace right now.  I think we should leave it until the marketplace has more options.

Reply:
Agreed.